torchgeo/trainers/change.py

```diff
@@ -235,7 +312,16 @@ def _shared_step(self, batch: Any, batch_idx: int, stage: str) -> Tensor:
                     keepdim=True,
                 )
                 batch = aug(batch)
-                batch['prediction'] = (y_hat.sigmoid() >= 0.5).long()
+                match self.hparams['task']:
+                    case 'binary' | 'multilabel':
+                        prediction = (y_hat.sigmoid() >= 0.5).long()
+                        # Restore channel dimension for plotting compatibility
+                        batch['prediction'] = prediction.unsqueeze(1)
```

Comment:
We currently do this for all 3 tasks, should this be outside the match statement? Or should it not apply to some tasks? In fact, why is it needed all of a sudden, I don't think anything should have changed in this PR?